9th Terraform Recipe - Kafka with MongoDB source connector

dewan-ahmed opened the pull request:
# What changed, and why it matters

Resolves https://github.com/aiven/devportal/issues/871

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:
```diff
@@ -0,0 +1,86 @@
+Describe the setup
+------------------
+
+Here is the Terraform recipe that will spin up an Aiven for Apache Kafka service with Kafka Connect enabled. This recipe will also create and configure a MongoDB source connector. 
```
Maybe we can add a link here to https://developer.aiven.io/docs/products/kafka/kafka-connect/howto/mongodb-poll-source-connector.html

dewan-ahmed replied:
Thanks. Good point. I'm adding the link when I first mention MongoDB source connector, i.e. in the first paragraph.

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:
```diff
@@ -0,0 +1,86 @@
+      kafka_rest    = true
+      kafka_version = "3.1"
+      kafka {
+        group_max_session_timeout_ms = 70000
```
why do we have this custom settings? Are they strictly needed?

dewan-ahmed replied:
They are not; removed. Good call.

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:
```diff
@@ -0,0 +1,86 @@
+  
+  * Since you have ``kafka_connect`` set to ``true`` under the ``kafka_user_config``, you don't need a standalone Aiven for Apache Kafka Connect service.
+  * You need to enable the ``auto_create_topics_enable`` flag so that the connector is able to create the topic on the Apache Kafka cluster.
+  * The automatically created topic name will be **DATABASE.COLLECTION** - ``sample_airbnb.listingsAndReviews`` in this example.
```
```suggestion
  * The automatically created topic name will be  the concatenation of ``database`` and ``collection`` parameters - ``sample_airbnb.listingsAndReviews`` in this example.
```

ftisiot reviewed (changes requested):
Good work, but I couldn't test. Do you have an existing MongoDB I can use for testing? Also left a couple comments

ftisiot commented:
The code is working (I can see the mongodb data in Kafka) I'm ready to have another look at the surroundings once you fix the above issues!

dewan-ahmed commented:
Thanks Francesco. I implemented the review feedback. Some notes:

1. I added MongDB connector link in the first paragraph where it's mentioned first.
2. I added a line that in case the DB is not reachable via public internet, it has to be part of a VPC. I think having a link so that the reader can learn more about VPC, would be helpful. But I couldn't find much info on DevPortal/Blog that I could link.

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:
```diff
@@ -0,0 +1,84 @@
+.. Tip:: 
+  
+  * Since you have ``kafka_connect`` set to ``true`` under the ``kafka_user_config``, you don't need a standalone Aiven for Apache Kafka Connect service.
+  * You need to enable the ``auto_create_topics_enable`` flag so that the connector is able to create the topic on the Apache Kafka cluster.
```
```suggestion
  * The ``auto_create_topics_enable`` flag is enabled, therefore the connector is able to create the topic on the Apache Kafka cluster by pushing the first message, without having to create the topic first.
```

dewan-ahmed commented:
Thanks Francesco. I added the Kafka Connect warning and reworded the topic auto creation sentence.

I removed the Tip from the explanation of the parameters since that section is not a tip. This also aligns with how other recipes are structured.

ftisiot reviewed (approved):
LGTM! Great work
